server/src/main.rs

```diff
@@ -749,6 +814,34 @@ fn trace_stream(res: &mut Response<Streaming>, child: &mut child::Thread) -> io:
                 api::Trace { index: 0, line: 0, data: data }
             }
 
+            DebugMessage::Trace(DebugTrace::StopRequest) => {
+                done = true;
+                let e_type = match child.execution {
+                    Some((_, e_type)) =>  e_type,
+                    _ => return Err(io::Error::new(io::ErrorKind::NotFound, "no such execution")),
+                };
```

Comment:
Since `child.execution` is unconditionally cleared below, this should use `child.execution.take()`. Since there is always an execution at this point, it should just `.unwrap()` rather than matching.